scripts/tests/run_tv_casting_test.py

```
@@ -0,0 +1,225 @@
+                if "Discovered Commissioner" in next_line:
+                    break
+
+            # Append the extracted information to the devices list
+            discovered_commissioners.append({
+                "discovered_commissioner": commissioner,
+                "vendor_id": vendor_id,
+                "product_id": product_id,
+                "device_type": device_type
+            })
+
+    # If the list of discovered commissioners is empty and we didn't find the "No commissioner discovered" string,
+    # then something went wrong. Exit on error.
+    if len(discovered_commissioners) == 0:
+        print('No commissioner(s) discovered! The list of discovered commissioner(s) is empty!')
+        handle_discovery_failure()
+
+
+# Test if the Linux tv-casting-app is able to discover the Linux tv-app.
+# The Linux tv-casting-app and the tv-app will be run in separate processes.
+# Their corresponding output will be written to their respective log files.
+# The log file of the tv-casting-app will be parsed for strings of interest
+# which will be printed to the console.
+def test_discovery_fn():
+
+    with open(LINUX_TV_APP_LOGS, 'w') as fd1, open(LINUX_TV_CASTING_APP_LOGS, 'w') as fd2:
+
+        # Run the Linux tv-app and write the output to file
+        tv_app_rel_path = 'out/tv-app/chip-tv-app'
+        tv_app_abs_path = os.path.abspath(tv_app_rel_path)
+        subprocess.Popen(tv_app_abs_path, stdout=fd1, stderr=subprocess.PIPE, text=True)
+
+        time.sleep(RUN_INTERVAL)
+
+        # Run the Linux tv-casting-app and write the output to file
+        tv_casting_app_rel_path = 'out/tv-casting-app/chip-tv-casting-app'
+        tv_casting_app_abs_path = os.path.abspath(tv_casting_app_rel_path)
+        tv_casting_app_process = subprocess.Popen(
+            tv_casting_app_abs_path, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+
+        for line in tv_casting_app_process.stdout:
```

Just like we are failing fast, we could succeed fast as well here by looking for the commissioner with the expected vid, pid, dt and succeeding accordingly.

---

Agreed, implementation has been updated to succeed fast after finding commissioner with expected vendor id, product id, and device type.